web-marketplace/src/legacy-pages/Dashboard/hooks/useUpdateOrganizationProfile.ts

```diff
@@ -0,0 +1,308 @@
+        'banner',
+        sanitizedBackgroundImage,
+        currentValues?.backgroundImage ?? null,
+      );
+      maybeAddMetadataAction(
+        'website_link',
+        sanitizedWebsite,
+        currentValues?.websiteLink ?? null,
+      );
+      maybeAddMetadataAction(
+        'twitter_link',
+        sanitizedTwitter,
+        currentValues?.twitterLink ?? null,
+      );
+
+      const executeActionsMsg = getExecuteActionsWasm(actions);
+      const executeMsg = getMsgExecuteContract({
+        walletAddress: wallet.address,
+        contract: rbamAddress,
+        executeActionsMsg,
+      });
+
+      const result = await signAndBroadcast({
+        msgs: [executeMsg],
+        fee: 'auto',
+      });
+
+      if (typeof result === 'string') {
+        throw new Error(result);
+      }
+
+      const maxAttempts = 15;
+      const pollInterval = 500;
+      let attempts = 0;
+      let indexerUpdated = false;
+
+      while (attempts < maxAttempts && !indexerUpdated) {
+        attempts++;
+        await timer(pollInterval);
+
+        try {
+          const data = await reactQueryClient.fetchQuery(
+            getAccountByIdQuery({
+              client: graphqlClient as ApolloClient<NormalizedCacheObject>,
+              enabled: !!graphqlClient && !!activeAccountId,
+              id: activeAccountId ?? '',
+              languageCode: selectedLanguage,
+            }),
+          );
+
+          const dao =
+            data?.accountById?.daosByAssignmentAccountIdAndDaoAddress?.nodes?.find(
+              node =>
+                node?.address === daoAddress ||
+                (organizationId &&
+                  node?.organizationByDaoAddress?.id === organizationId),
+            );
+
+          const organization = dao?.organizationByDaoAddress;
+
+          const nameMatches = organization?.name === sanitizedName;
+
+          const descriptionMatches =
+            organization?.description === sanitizedDescription;
+
+          const profileImageMatches =
+            organization?.image === sanitizedProfileImage;
+
+          const backgroundImageMatches =
+            organization?.bgImage === sanitizedBackgroundImage;
+
+          const websiteMatches = organization?.websiteLink === sanitizedWebsite;
+
+          const twitterMatches = organization?.twitterLink === sanitizedTwitter;
+          if (
+            nameMatches &&
+            descriptionMatches &&
+            profileImageMatches &&
+            backgroundImageMatches &&
+            websiteMatches &&
+            twitterMatches
+          ) {
+            indexerUpdated = true;
+          }
+        } catch (error) {
+          continue;
+        }
+      }
+
+      await reactQueryClient.invalidateQueries({
+        queryKey: getAccountByIdQuery({
+          client: graphqlClient,
+          enabled: !!graphqlClient && !!activeAccountId,
+          id: activeAccountId ?? '',
+          languageCode: selectedLanguage,
+        }).queryKey,
+      });
```

Comment:
this is not needed, since you're already refetching the account data until you see the org updates
so the latest data will already be populated in the cache